utils/utils.go

```diff
@@ -3,40 +3,86 @@ package utils
 import (
 	"strings"
 
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
 	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/vishvananda/netlink"
 )
 
 const (
-	hostLabelKeyword = "__host_label__"
+	hostLabelKeyword     = "__host_label__"
+	hostInterfaceKeyword = "__host_interface__"
```

Comment:
We already have the ability to replace any host label, why do we need this? I don't think this is correct.

Example usage:
`bridgeSubnet: "__host_label__: io.rancher.network.per_host_subnet.subnet"`
[https://github.com/leodotcloud/test-per-host-subnet-catalog-item/blob/master/infra-templates/per-host-subnet/0/docker-compose.yml.tpl#L30]

Similarly you need:
`bridgeIP: '__host_label__: io.rancher.network.l2_flat.bridge_ip'`

Reply:
@leodotcloud 
From the code point of view, we do not need to introduce a new keyword `__host_interface`,
but from the ease of use point of view, the user has configured the flat interface in the catalog options, why also allow users to configure a host label.

Reply:
@leodotcloud ,

@niusmallnan explained it to me.

Without the new keyword, the user will have to:
1. Go to each host, figure out what's the IP for the bridge interface.
2. Add label `io.rancher.network.l2_flat.bridge_ip` to each host manually, with correlated IPs.

This is a quite unfriendly user experience.

With the new keywords, the user doesn't need to do anything. It will just work. Because `__host_interface__` will trigger an automatic IP replacement. Plugin manager will find the IP on that interface, and insert it to the `bridgeIP` in the CNI config automatically.

User has to provide information for `per-host-subnet`, because it's unknown to us. Here is different. Here we know the IP and we can fill it for the user.